### PR TITLE
Set MSVS environment variables

### PR DESCRIPTION
**Pull request by vsebe**
Signed-off-by: Violeta Sebe <vsebe@ca.ibm.com>

### CODE REVIEW

File: closed/OpenJ9.gmk [hunk 1]
```diff
@@ -333,11 +336,13 @@ run-preprocessors-j9 : stage-j9 \
 build-j9 : run-preprocessors-j9
 	@$(ECHO) Compiling OpenJ9 in $(OUTPUT_ROOT)/vm
 ifeq ($(OPENJ9_ENABLE_CMAKE),true)
-	(export OMR_DIR=$(OUTPUT_ROOT)/vm/omr OPENJ9_BUILD=true && \
+	(export OPENJ9_BUILD=true $(EXPORT_MSVS_ENV_VARS) && \
```
Comment (pshipton) [inline closed/OpenJ9.gmk]: Did you mean to remove ```OMR_DIR=$(OUTPUT_ROOT)/vm/omr```

**Comment by vsebe** [comment]
After the Microsoft Visual Studio is installed the environment variables are not set. The user has to add VS to the path and set LIB and INCLUDE. This is not required when building openjdk9 with hotspot.
To avoid this extra step when setting the environment , there variables are set in the makefile.

Administrator@win7x64rt1 /cygdrive/c/temp/openj9-openjdk-jdk9
$ build/windows-x86_64-normal-server-release/images/jdk/bin/java -version
openjdk version "9-internal"
OpenJDK Runtime Environment (build 9-internal+0-adhoc.Administrator.openj9-openjdk-jdk9)
Eclipse OpenJ9 VM (build 2.9, JRE 9 Windows 7 amd64-64 Compressed References 20171109_000000 (JIT enabled, AOT enabled)
OpenJ9   - 4711e38
OMR      - e1404d50
OpenJDK  - f7a4f94568 based on jdk-9+181)

**Comment by vsebe** [comment]
@jenkins test sanity plinux

File: closed/OpenJ9.gmk (hunk 1 above)
Comment (vsebe) [reply closed/OpenJ9.gmk]: Yes, I did. It is not required. @dnakamura please confirm, thank you!

File: closed/OpenJ9.gmk [hunk 2]
```diff
@@ -96,11 +96,14 @@ else
 endif
 
 ifeq ($(OPENJDK_TARGET_OS),windows)
+  # set Visual Studio environment
+  EXPORT_MSVS_ENV_VARS := PATH="$(PATH)" INCLUDE=$(INCLUDE) LIB=$(LIB)
```
Comment (keithc-ca) [inline closed/OpenJ9.gmk]: The values of INCLUDE and LIB will also likely need to be enclosed in double-quotes.

File: closed/OpenJ9.gmk [hunk 3]
```diff
@@ -333,11 +336,13 @@ run-preprocessors-j9 : stage-j9 \
 build-j9 : run-preprocessors-j9
 	@$(ECHO) Compiling OpenJ9 in $(OUTPUT_ROOT)/vm
 ifeq ($(OPENJ9_ENABLE_CMAKE),true)
-	(export OMR_DIR=$(OUTPUT_ROOT)/vm/omr OPENJ9_BUILD=true && \
+	(export OPENJ9_BUILD=true $(EXPORT_MSVS_ENV_VARS) && \
 		cd $(OUTPUT_ROOT)/vm/build && $(MAKE) $(MAKEFLAGS) install \
 	)
 else
-	(export OPENJ9_BUILD=true $(EXPORT_NO_USE_MINGW) && cd $(OUTPUT_ROOT)/vm && $(MAKE) $(MAKEFLAGS) all)
+	(export OPENJ9_BUILD=true $(EXPORT_NO_USE_MINGW) $(EXPORT_MSVS_ENV_VARS) && \
```
Comment (keithc-ca) [inline closed/OpenJ9.gmk]: Please split these lines consistently with other command sequences (`&&` at the start of a line with the next command).

File: closed/OpenJ9.gmk (hunk 2 above)
Comment (keithc-ca) [reply closed/OpenJ9.gmk]: On second reading, I see those variables get their quotes from custom-spec.gmk. A comment noting that difference would be helpful.

Comment (vsebe) [reply closed/OpenJ9.gmk]: I added comments

File: closed/OpenJ9.gmk (hunk 3 above)
Comment (vsebe) [reply closed/OpenJ9.gmk]: Fixed!

**Comment by keithc-ca** [comment]
jenkins test sanity

File: closed/OpenJ9.gmk [hunk 4]
```diff
@@ -333,11 +338,15 @@ run-preprocessors-j9 : stage-j9 \
 build-j9 : run-preprocessors-j9
 	@$(ECHO) Compiling OpenJ9 in $(OUTPUT_ROOT)/vm
 ifeq ($(OPENJ9_ENABLE_CMAKE),true)
-	(export OMR_DIR=$(OUTPUT_ROOT)/vm/omr OPENJ9_BUILD=true && \
-		cd $(OUTPUT_ROOT)/vm/build && $(MAKE) $(MAKEFLAGS) install \
+	(export OPENJ9_BUILD=true $(EXPORT_MSVS_ENV_VARS) \
```
Comment (keithc-ca) [inline closed/OpenJ9.gmk]: The trailing whitespace (after the `\`) must be removed to avoid breaking cmake builds.

File: closed/OpenJ9.gmk (hunk 1 above)
Comment (dnakamura) [reply closed/OpenJ9.gmk]: Correct. it is a vestigial variable

File: closed/OpenJ9.gmk (hunk 4 above)
Comment (vsebe) [reply closed/OpenJ9.gmk]: Spaced removed, thank you!

**Pull request merged**